lib/shim/message-shim.js

```diff
@@ -750,6 +750,12 @@ function recordSubscribedConsume(nodule, properties, spec) {
               ATTR_DESTS.NONE,
               'message.parameters.' + key,
               msgDesc.parameters[key])
+
+            tx.baseSegment.attributes.addAttribute(
```

Comment:
I felt it would be more efficient than allocating a segment from `shim.agent.tracer.getSegment()`.

Reply:
i wouldn't expect it to be allocating anything. and while a function call will be slightly less efficient... we aren't talking thousands per transaction but rather a few per transaction. as such, i don' think it is hot enough path for function call VS `.member` access to be a concern. it might be if there was actually an object allocation.

that being said, i don't know if `baseSegment` or what is in context there. there *could* be a good reason for doing so, but i'd double-check and not explicitly do that if not... since we are trying to get away from that dump everything on one span model. if there does end up being a need, we should prob comment that need.

Reply:
For this one, we use `shim.createSegment` to set the `baseSegment`.  It doesn't seem like I should create a new segment here.

Reply:
why would we be creating a new segment? 

we should be grabbing the 'current' segment... which might still end up as `baseSegment` unless there's a reason not to. 

`tracer.getSegment()`, which you mention above, does not create a segment.

Reply:
ok, avoiding getting tripped up on teh "create" part of this... it sounds like what is being implied is we "just" created a new segment. and while it may not be in context, it may make sense to set attributes on there.

as such, i'd agree.

Reply:
Do you think `shim.getActiveSegment()` would me more appropriate here?

Reply:
well, *here*... based on your comments i see we've just created a segment so i'm fine with it going on that just created segment. i got caught up by it being immediately assigned to `baseSegment` and missed that. and then distracted by comments about creating. so it turns out ultimately, there was a good reason for adding to that segment / seems fine as that segment won't yet have been assigned as the current segment.

in other locations, where that is not the case, i would grab the active segment.

`shim` likely may not apply because it may not be at that outer instrumentation level. `shim.getActiveSegment()` ultimately calls `tracer.getSegment()` unless the segment was assigned to the object passed in. so `tracer.getSegment()` will be fine for non-shim use cases.